src/web3-contracts.js

```diff
@@ -253,28 +257,64 @@ export function useConvertAntToAnj() {
   )
 }
 
-// export function useConvertEthToAnj() {
-// const { account } = useWeb3Connect()
-// const [wrapperAddress] = getKnownContract('WRAPPER')
-// // for making this line usable we need to have the ABI
-// const wrapperContract = useContract(wrapperContract, wrapperAbi)
-
-// return useCallback(
-// async ethAmount => {
-// if (!wrapperContract) {
-// return false
-// }
-// const tx = await wrapperContract.methods
-// .contributeEth(0, 60, true)
-// .send({ from: account, value: ethAmount })
-// if (tx.error) {
-// return false
-// }
-// return true
-// },
-// [account]
-// )
-// }
+export function useConvertTokenToAnj(selectedToken) {
+  const tokenContract = useKnownContract(`TOKEN_${selectedToken}`)
+  const wrapperContract = useKnownContract(`WRAPPER`)
+  const [tokenAddress] = getKnownContract(`TOKEN_${selectedToken}`)
+  const [wrapperAddress] = getKnownContract('WRAPPER')
+
+  return useCallback(
+    async (amount, estimatedAnj) => {
+      if ((!tokenContract && selectedToken !== 'ETH') || !wrapperAddress) {
+        throw new Error('Could not get the token and wrapper contract.')
+      }
+
+      // If the user has selected ETH, we can just send the ETH to the function
+      const fiveMinutes = Math.floor(Date.now() / 1000) + 60 * 5
+
+      if (selectedToken === 'ETH') {
+        return await wrapperContract.contributeEth(1, fiveMinutes, true, {
+          gasLimit: 1000000,
+          value: bigNum(amount),
+        })
+      }
+
+      // If the user has selected ANT, we can directly
+      // approve and call the wrapper using ANT's approveAndCall
+      if (selectedToken === 'ANT') {
+        return tokenContract.approveAndCall(wrapperAddress, amount, '0x00', {
+          gasLimit: 1000000,
+        })
+      }
+      // else, we will need two transactions: the approval,
+      const approval = await tokenContract.approve(wrapperAddress, amount, {
```

Comment:
See https://github.com/aragon/anj.aragon.org/pull/28/files#r373419565; we should check the account's current allowance and if they've already added enough allowance from before, skip this transaction.